### PR TITLE
Remove the outdated information from the docs

### PR DESCRIPTION
**Pull request by lubosmj**

### CODE REVIEW

**Comment by lubosmj** [comment]
Inspired by https://github.com/pulp/pulp-cli/blob/main/releasing.md (which is easy to find and follow).

File: MANIFEST.in [hunk 1]
```diff
@@ -15,3 +15,4 @@ include COPYRIGHT
 include pulpcore/app/templates/rest_framework/api.html
 include manage.py
 include test_requirements.txt
+include RELEASING.md
```
Comment (mdellweg) [inline MANIFEST.in:18]: Actually, let's ignore and not include this file in the published artifacts.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Releasing (For Internal Use)
+
+### Check If the Release is Necessary
+- Run the release checker script:
+  ```
+  python3 .ci/scripts/check_release.py
+  ```
+
+### Create a New Y-branch of Pulpcore (e.g., 3.23)
+- If needed, trigger a [Create New Release Branch](https://github.com/pulp/pulpcore/actions/workflows/create-branch.yml) job via the GitHub Actions.
```
Comment (mdellweg) [inline RELEASING.md]: This creates the bump version PR. That one needs to be watched and prioritised.

File: pulp_file/app/viewsets.py [hunk 3]
```diff
@@ -535,8 +535,6 @@ class FileDistributionViewSet(DistributionViewSet, RolesMixin):
 class FileAlternateContentSourceViewSet(AlternateContentSourceViewSet, RolesMixin):
     """
     Alternate Content Source ViewSet for File
-
-    ACS support is provided as a tech preview in pulp_file.
```
Comment (mdellweg) [inline pulp_file/app/viewsets.py:539]: These changes warrant for at least a separate commit.

**Review by mdellweg — COMMENTED** [review]
I'm a fan of keeping the information with the code. In this case, I guess the "release.md" should come with the template.

File: pulp_file/app/viewsets.py (hunk 3 above)
Comment (lubosmj) [reply pulp_file/app/viewsets.py:539]: I found these to be a leftover after merging pulp-file into pulpcore.

**Comment by lubosmj** [comment]
> I'm a fan of keeping the information with the code. In this case, I guess the "release.md" should come with the template.

Good idea. However, how can we ensure that this file is shipped together with the `plugin-template plugin --github` update? It will come from the root directory of the plugin's template.

**Comment by mdellweg** [comment]
I think it should go in the templated section. The plugin template repo does not want to have a "How to release" document for itself.

**Comment by lubosmj** [comment]
The initial change should be now shipped through the plugin template, please review: https://github.com/pulp/plugin_template/pull/919.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**